mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/scheduler/MantisSchedulerFactoryImpl.java

```diff
@@ -39,17 +36,15 @@ public class MantisSchedulerFactoryImpl implements MantisSchedulerFactory {
     private final ResourceClusters resourceClusters;
     private final ExecuteStageRequestFactory executeStageRequestFactory;
     private final JobMessageRouter jobMessageRouter;
-    private final SchedulingService mesosSchedulingService;
     private final MasterConfiguration masterConfiguration;
     private final MetricsRegistry metricsRegistry;
     private final Map<ClusterID, MantisScheduler> actorRefMap = new ConcurrentHashMap<>();
 
     @Override
-    public MantisScheduler forClusterID(@Nullable ClusterID clusterID) {
-        Optional<ClusterID> clusterIDOptional = Optional.ofNullable(clusterID);
-        if (clusterIDOptional.isPresent()) {
-            if (Strings.isNullOrEmpty(clusterIDOptional.get().getResourceID())) {
-                log.error("Received empty resource id: {}", clusterIDOptional.get());
+    public MantisScheduler forClusterID(ClusterID clusterID) {
+        if (clusterID != null) {
```

Comment:
when will this be null given that the parameter is not marked nullable?

Reply:
never. (so null case is just log and throw).